component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java

```diff
@@ -361,24 +366,52 @@ private boolean isValidSignature(String clientId, SignedJWT signedJWT, String te
                                      String alias) throws OAuthClientAuthnException {
 
         X509Certificate cert = null;
+        String jwksUri = "";
+        boolean isValidSignature = false;
         try {
             cert = (X509Certificate) OAuth2Util.getX509CertOfOAuthApp(clientId, tenantDomain);
         } catch (IdentityOAuth2Exception e) {
             if (log.isDebugEnabled()) {
                 log.debug(e.getMessage(), e);
```

Comment:
e anyways contains e.getMessage. Hence let's use a more specific text as the message.

Reply:
added a message